plugins/modules/monitoring/newrelic_deployment.py

```diff
@@ -99,47 +86,62 @@ def main():
             application_id=dict(required=False),
             changelog=dict(required=False),
             description=dict(required=False),
-            revision=dict(required=False),
+            revision=dict(required=True),
             user=dict(required=False),
-            appname=dict(required=False),
-            environment=dict(required=False),
-            validate_certs=dict(default=True, type='bool'),
         ),
-        required_one_of=[['app_name', 'application_id']],
-        supports_check_mode=True
+        required_one_of=[['app_name', 'application_id']]
     )
 
-    # build list of params
-    params = {}
-    if module.params["app_name"] and module.params["application_id"]:
-        module.fail_json(msg="only one of 'app_name' or 'application_id' can be set")
-
-    if module.params["app_name"]:
-        params["app_name"] = module.params["app_name"]
-    elif module.params["application_id"]:
-        params["application_id"] = module.params["application_id"]
+    if module.params['app_name'] and module.params['application_id']:
+        module.fail_json(msg="both app_name' and 'application_id'\
+        are defined")
+
+    if module.params['app_name']:
+        data = 'filter[name]=' + str(module.params['app_name'])
```

Comment:
Please use a proper encoding function for forms. Since you are using `fetch_url`, you can simply pass a dictionary and it will take care of encoding automatically (see https://github.com/ansible-collections/community.hrobot/blob/main/plugins/module_utils/failover.py#L64).

Reply:
Now, I don't have access to NewRelic so I can't test it.
is this a mandatory change?

Reply:
You need to fix a proper method which will encode values correctly. Simple string concatenation is a disaster waiting to happen (in the worst case, a security vulnerability).

Reply:
I can't modify something which I can't test, module was tested and still in use.
If you say, without modify this it can't be merged than so let it be.

Reply:
That's unfortunate. In case you don't change this to use `urlencode` or simply provide a dict, I will not merge this. Sorry.

Reply:
**Note** 
if this was so important change, Others/You as Reviewer should have pointed out in my first PR & second PR which was there from 2018 but None. 
Now you want this to changed. **Hell No!**

I will not change any working piece of code which can break the new module implementation. As I don't have access to newrelic **to test your enforced changes** which means **deadend**.

Have Fun!

Reply:
This is a very trivial change to make, which does not affect the whole module. So I see absolutely no need to point out such problems as early as possible in the review process. And if you don't want to make any changes to the code's behavior, well, then I don't see why a review makes sense anyway.

I've now spend quite some time reviewing your code and changes in my free time. If you are not interested in getting this included, then I guess I totally wasted that time.

Reply:
what about my time man? do you understand the level of patience from my side ( 2 years ) ?

it's not about trivial change, it's about basic testing, if I can't confirm as maintainer that it will work or not with urlencode/dict then I how i can even make a change, I don't understand that part.

If you have access to NewRelic please make the test with urlencode/dict and suggest the actual code change, I happy to update the code.

**Just for your knowledge,** half of module will stop working if this data is passed in wrong format as the search api will return none and anyone using `app_name` as parameter will never work until someone fix it in future.

let's assume your case where it cause some security Issue then it will on NewRelic Side why you/ansible need to worry. NewRelic should not prevent/fix this in there API as a validation and later Ansible Community should fix as part of bugfix process. 
Now when I don't have testing resources/credentials, On what base you will merge it even if I change it?